packages/dev/core/src/DeviceInput/InputDevices/deviceSourceManager.ts

```diff
@@ -91,12 +78,18 @@ export class DeviceSourceManager implements IDisposable, IObservableManager {
         this._engine._deviceSourceManager._refCount++;
 
         // Observables
-        this.onDeviceConnectedObservable = new Observable<DeviceSource<DeviceType>>((observer) => {
-            this.getDevices().forEach((device) => {
-                this.onDeviceConnectedObservable.notifyObserver(observer, device);
-            });
+        this.onDeviceConnectedObservable = new Observable((observer) => {
+            for (const devices of this._devices) {
+                if (devices) {

```

Comment:
Is it possible for this to be missing?

Reply:
With gamepads I believe that this is possible

Reply:
Can you confirm? It seems weird to have a map with null entries.

Reply:
I can confirm this.  The `_devices` object is not technically a map but just an array of an array.  When we first instantiate it, we just create an entry for each and every DeviceType.  This means that there could be devices that haven't initialized their arrays (eg. _devices[DeviceType.Keyboard]).  If all else fails, we _could_ convert it to a Map, which would remove the need to check if a DeviceSource array for a specific DeviceType exists.

Reply:
Okay, sounds good.